spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/intermediate/IRConverter.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.utils.intermediate
+
+import com.intel.analytics.bigdl.nn.Graph
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.nn.mkldnn.{DnnGraph, InputWrapper, Output}
+import com.intel.analytics.bigdl.tensor.{FloatType, Tensor}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.{Module, utils}
+import com.intel.analytics.bigdl.utils.{MklBlas, MklDnn, Node}
+
+import scala.reflect.ClassTag
+
+
+private[bigdl] class IRConverter[T: ClassTag](IRgraph: IRGraph[T])(implicit ev: TensorNumeric[T]) {
+  /**
+   * convert IRgraph to blas or dnn graph.
+   * @param dnnMode If dnnMode = true, it means ir graph must be converted to dnn graph
+   * @return
+   */
+  def toGraph(dnnMode: Boolean = false) : Graph[T] = {
+    if (utils.Engine.getEngineType() == MklBlas && !dnnMode) {
```

Comment:
is underlying engine related to conversion ?

Reply:
I see you check the engine type when converting to BLAS, but you don't check when converting to DNN, instead you just   set the engine type to DNN for DNN conversion

Reply:
ok, delete dnnMode and do some change